scripts/synapse_port_db

```diff
@@ -820,7 +820,7 @@ class Porter(object):
                     "ALTER SEQUENCE events_stream_seq RESTART WITH %s", (next_id,)
                 )
 
-            txn.execute("SELECT -MIN(stream_ordering) FROM events")
+            txn.execute("SELECT GREATEST(-MIN(stream_ordering), 1) FROM events")
```

Comment:
I assume this does fix things, but what is the purpose of negating the minimum `stream_ordering` here? And wouldn't using `GREATEST` and `-` here mean that we almost always end up with 1 as it's greater?

Reply:
`stream_ordering` is negative for backfilled events, so if e.g. the min is -52, then `GREATEST(- -52 1)` is `52`, while if there are no backfilled events we may end up with a positive min so we get `GREATEST(-1, 1)` which is `1`.